core/src/test/java/ai/timefold/solver/core/impl/heuristic/selector/move/generic/RuinRecreateMoveTest.java

```diff
@@ -0,0 +1,95 @@
+package ai.timefold.solver.core.impl.heuristic.selector.move.generic;
+
+import static ai.timefold.solver.core.impl.testdata.util.PlannerTestUtils.mockRebasingScoreDirector;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+import static org.mockito.Mockito.mock;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import ai.timefold.solver.core.impl.domain.variable.descriptor.GenuineVariableDescriptor;
+import ai.timefold.solver.core.impl.solver.scope.SolverScope;
+import ai.timefold.solver.core.impl.testdata.domain.TestdataEntity;
+import ai.timefold.solver.core.impl.testdata.domain.TestdataSolution;
+import ai.timefold.solver.core.impl.testdata.domain.TestdataValue;
+
+import org.junit.jupiter.api.Test;
+
+class RuinRecreateMoveTest {
```

Comment:
Should we add multithreading tests using the R&R moves?

Reply:
I was thinking about it, but the only way to do that would be to add (another) test that runs the entire solver. I don't want to do that - those tests don't actually test anything specific and they take too long. We should have less of those, not more.